### PR TITLE
import remaining blog posts from the labs repository. #43 

### PR DESCRIPTION
**Pull request by makkoncept**
**Description**

I have used [this python script](https://gist.github.com/makkoncept/45303a19fc8e8ee45d7f96555e7e9e76) to automate the conversion of HTML files(in [_posts](https://github.com/creativecommons/labs/tree/gh-pages/_posts)) to corresponding directories with contents.lr file inside. The script looks for the images referenced in HTML files in [img](https://github.com/creativecommons/labs/tree/gh-pages/img) directory and copies them to the folder. The files whose images were not find are printed to the console so that we can check manually in other places. Two markdown files were remaining which were ported manually. 

Images corresponding to the following html files are not available:

- _posts/2012-02-22-cc-legaleratta-errata-annotations-without-republishing-licenses.html
- _posts/2007-05-01-stuffit-10-considered-harmful.html
- _posts/2007-08-03-licensing-on-the-gnome-desktop.html
- _posts/2007-08-03-licensing-on-the-gnome-desktop.html
- _posts/2007-08-03-licensing-on-the-gnome-desktop.html
- _posts/2007-05-04-palimpsest.html
- _posts/2007-06-21-enhanced-metadata-graduates-from-labs.html
- _posts/2010-07-11-gsoc-firefox-extension-status-report.html
- _posts/2011-06-27-powerofopen-metrics.html
- _posts/2007-06-27-its-coming-cc-openofficeorg-add-in.html
- _posts/2007-06-27-its-coming-cc-openofficeorg-add-in.html

**Other information**

**Checklist:**
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by kgodey** [comment]
Also, has the `blog-post.ini` model been updated with the new field for `categories`? I can't find it in the diff but GitHub also isn't loading all files since it's so big.

**Comment by makkoncept** [comment]
No, I thought I would do that after the draft PR would merge. 

Just to confirm adding the following would do?
```
[fields.categories]
label = Categories
type = checkboxes
```

**Comment by kgodey** [comment]
I would use `strings` as the type since `checkboxes` needs to have a source. strings would require each category to be on a separate line though.
https://www.getlektor.com/docs/api/db/types/strings/

**Comment by makkoncept** [comment]
category need to be on a separate line in `<post_name>/content.lr` for string type?

**Comment by kgodey** [comment]
I believe so, I haven't used it before and I'm going off the documentation I linked above.

**Comment by makkoncept** [comment]
I can make a script to do it.
Just to clarify, the format must be:
```
title: 32 to 64 bit remotely
---
categories: 
Debian
software
---
author: nkinkade
---
```

**Comment by kgodey** [comment]
Yeah, that looks right. You could test it by trying to read the values in the template like the documentation shows.

**Comment by makkoncept** [comment]
yes, I have tested it using the Lektor documentation example and it is working.
I will make the script to automate this.

**Comment by makkoncept** [comment]
@kgodey I have done it. Please check the files. 
[Here](https://gist.github.com/makkoncept/edcb22a2754aa1074d4258ee8d990b59) is the script for reference.

File: content/tech-blog-archives/getting-started-with-hacking-on-the-list/contents.lr [hunk 1]
```diff
@@ -1,6 +1,7 @@
 title: Getting started with hacking on The List
 ---
-author: Matt Lee
+categories:
+Lee
```
Comment (kgodey) [inline content/tech-blog-archives/getting-started-with-hacking-on-the-list/contents.lr]: There's some kind of error in the script. This post should have "development" as a category and nothing else (see: original file: https://raw.githubusercontent.com/creativecommons/labs/gh-pages/_posts/2014-12-15-getting-started-hacking-on-the-list.md)

Comment (makkoncept) [reply content/tech-blog-archives/getting-started-with-hacking-on-the-list/contents.lr]: This is the only file with this error. I forgot to add the categories to it after it was merged in #43 and therefore the script not worked properly on it. I will correct it.

**Comment by kgodey** [comment]
this looks good, I've verified with @makkoncept on Slack that the number of blog posts ported over match up.

**Pull request merged**